doc/api/deprecations.md

```diff
@@ -2617,6 +2617,49 @@ async function openAndClose() {
 }
 ```
 
+<a id="DEP0XXX"></a>
+### DEP0XXX: `module.parent`
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/32217
+    description: Runtime deprecation.
+-->
+
+Type: Runtime
+
+A CJS module can access the first module that required it using `module.parent`.
+This feature does not work when a module is imported using ECMAScript modules
+specification, therefore it is deprecated.
+
+Some modules use it to check if they are the entry point of the current process.
+Instead, it is recommended to compare `require.main` and `module`:
+
+```js
+if (require.main === module) {
+  // CLI code runs here
+}
+```
+
+When looking for the CJS modules that have required the current one,
+`module.children` can be used:
+
+```js
+const moduleParents = [];
+function findCurrentModuleParents(moduleParent = require.main) {
```

Comment:
this code will form an infinite loop in the case of circular dependencies. I would recommend doing something like this:
```js
const parents = Object.values(require.cache)
  .filter((m) => m.children.includes(module));
```

Reply:
That's way more concise and readable, thanks!